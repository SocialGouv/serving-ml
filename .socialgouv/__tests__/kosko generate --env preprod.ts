//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/github-actions.env";

jest.setTimeout(1000 * 60);
test("kosko generate --preprod", async () => {
  // process.env.HARBOR_PROJECT = "cdtn";
  expect(
    await getEnvManifests("preprod", "", {
      ...project("serving-ml").preprod,
      PREPRODUCTION: "true",
      RANCHER_PROJECT_ID: "c-bd7z2:p-wcxt4",
      SOCIALGOUV_BASE_DOMAIN: "dev.fabrique.social.gouv.fr"
    })
  ).toMatchSnapshot();
});
