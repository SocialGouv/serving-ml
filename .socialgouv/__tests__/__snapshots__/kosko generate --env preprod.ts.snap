// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --preprod 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    socialgouv/creator: autodevops
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: c-bd7z2:p-wcxt4
    git/branch: refs/tags/v1.2.3
    git/remote: socialgouv/serving-ml
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/serving-ml
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    azure-pg-admin-user: serving-ml
    application: serving-ml
    component: serving-ml
    owner: serving-ml
    team: serving-ml
    cert: wildcard
  name: serving-ml-preprod
---
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: netpol-serving-ml-preprod
  namespace: serving-ml-preprod
spec:
  ingress:
    - from:
        - podSelector: {}
    - from:
        - namespaceSelector:
            matchLabels:
              network-policy/source: ingress-controller
    - from:
        - namespaceSelector:
            matchLabels:
              network-policy/source: monitoring
  podSelector: {}
  policyTypes:
    - Ingress
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/serving-ml
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: serving-ml
    application: serving-ml
    component: serving-ml
    owner: serving-ml
    team: serving-ml
    cert: wildcard
  name: serving-ml
  namespace: serving-ml-preprod
spec:
  replicas: 1
  selector:
    matchLabels:
      app: serving-ml
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.github.com/job: '5678'
        app.github.com/ref: refs/tags/v1.2.3
        app.github.com/repo: socialgouv/serving-ml
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: serving-ml
        application: serving-ml
        component: serving-ml
        owner: serving-ml
        team: serving-ml
        cert: wildcard
    spec:
      containers:
        - image: >-
            ghcr.io/socialgouv/cdtn/serving-ml:preprod-0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /v1/models/sentqam
              port: http
            initialDelaySeconds: 15
            periodSeconds: 5
            timeoutSeconds: 15
          name: serving-ml
          ports:
            - containerPort: 8501
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /v1/models/sentqam
              port: http
            initialDelaySeconds: 5
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 3
          resources:
            limits:
              cpu: 1000m
              memory: 2Gi
            requests:
              cpu: 500m
              memory: 1.5Gi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /v1/models/sentqam
              port: http
            periodSeconds: 5
            initialDelaySeconds: 0
            timeoutSeconds: 15
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: serving-ml
    application: serving-ml
    component: serving-ml
    owner: serving-ml
    team: serving-ml
    cert: wildcard
  name: serving-ml
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/serving-ml
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: serving-ml-preprod
spec:
  ports:
    - name: http
      port: 80
      targetPort: 8501
  selector:
    app: serving-ml
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/serving-ml
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: serving-ml
    application: serving-ml
    component: serving-ml
    owner: serving-ml
    team: serving-ml
    cert: wildcard
  name: serving-ml
  namespace: serving-ml-preprod
spec:
  rules:
    - host: serving-ml-preprod.dev.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: serving-ml
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - serving-ml-preprod.dev.fabrique.social.gouv.fr
      secretName: wildcard-crt
---
apiVersion: autoscaling/v2beta2
kind: HorizontalPodAutoscaler
metadata:
  name: serving-ml
  namespace: serving-ml-preprod
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/tags/v1.2.3
    app.github.com/repo: socialgouv/serving-ml
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: serving-ml
    application: serving-ml
    component: serving-ml
    owner: serving-ml
    team: serving-ml
    cert: wildcard
spec:
  maxReplicas: 10
  metrics:
    - resource:
        name: cpu
        target:
          averageUtilization: 100
          type: Utilization
      type: Resource
    - resource:
        name: memory
        target:
          averageUtilization: 100
          type: Utilization
      type: Resource
  minReplicas: 1
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: serving-ml
"
`;
